app/js/app/directives/react_classes/QuestionBlock.js

```diff
@@ -129,7 +146,12 @@ define(["react", "jquery"], function(React,$) {
 				// newVal must be a doc
 				var oldDoc = this.props.doc;
 				var newDoc = $.extend({}, oldDoc);
-				newDoc[key] = e.target.checked;
+				if (key === "requireUnits" && e.target.checked) {
+					newDoc[key] = e.target.checked;
+					newDoc["displayUnit"] = null;
+				} else {
+					newDoc[key] = e.target.checked;
+				}
```

Comment:
In both of the conditional branches you have `newDoc[key] = e.target.checked;` - it might be better to factor this out of the if-else, and that way you remove the else branch and make it 3 lines shorter? This change would also make the logic clearer (for me at least).